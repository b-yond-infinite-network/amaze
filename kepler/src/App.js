import React, {Component} from 'react';
import axios from 'axios';
import { BrowserRouter as Router, Route, Redirect} from "react-router-dom";

import HomePage from './_pages/HomePage';
import LoginPage from './_pages/LoginPage';
import PioneerPage from './_pages/PioneerPage';
import VariationPage from './_pages/VariationPage';
import ResourcePage from './_pages/ResourcePage';
import AdministrationPage from './_pages/AdministrationPage';

import './App.css';

const ping = {

  lastPing : false,

  lastTimeStamp : '',
  lastTimeStampUnit : '',

  timerPing : null,
  timerInterval : 5000,
}

const authentificationObtained = {

  isAuthenticated: false,

  isPioneer: false,
  isChecker: false, 
  isAdmin: false,

  pioneer: undefined,

  getIsAuthentificated() {
    return this.isAuthenticated;
  },
  reset() {

    this.isAuthenticated = false;

    this.isPioneer = false;
    this.isChecker = false;
    this.isAdmin = false;

    this.pioneer = undefined;
  },
  authenticate(level) {

    this.isAuthenticated = true;

    if (level >= 1) {
      this.isPioneer = true;
    }
    if (level >= 2) {
      this.isChecker = true;
    }
    if (level >= 3) {
      this.isAdmin = true;
    }
  },
  logout() {

    this.isAuthenticated = false;

    this.isPioneer = false;
    this.isChecker = false;
    this.isAdmin = false;
  }
}

const PioneerRoute = ({ component: Component, ...rest }) => (
  <Route {...rest} render={(props) => (authentificationObtained.isAuthenticated && authentificationObtained.isPioneer
      ? <Component {...props} />
      : <Redirect to={{
        pathname: '/',
        state: {tried: true}}}/>
  )} />
)

const CheckerRouter = ({ component: Component, ...rest }) => (
    <Route {...rest} render={(props) => (authentificationObtained.isAuthenticated && authentificationObtained.isChecker
      ? <Component {...props} />
      : <Redirect to={{
        pathname: '/',
        state: {tried: true}}}/>
  )} />
)

const AdministratorRouter = ({ component: Component, ...rest }) => (
  <Route {...rest} render={(props) => (authentificationObtained.isAuthenticated && authentificationObtained.isAdmin
      ? <Component {...props} />
      : <Redirect to={{
        pathname: '/',
        state: {tried: true}}}/>
  )} />
)

class App extends Component  {

  constructor (props) {
    super(props);

    /* bind the handle function to the object -> IMPORTANT */
    this.checkServerPing = this.checkServerPing.bind(this);
  }

  checkServerPing() {
    
    axios
      .get("http://localhost:8080/server/ping")
      .then(function(reponse) {

        ping.lastPing = true;
        ping.lastTimeStamp = reponse.data.timestamp;
        ping.lastTimeStampUnit = reponse.data.timestamp_unit;

      })
      .catch(error => {

        ping.lastPing = false;
        ping.lastTimeStamp = '';
        ping.lastTimeStampUnit = '';

        console.log(error)
      });
  }

  componentDidMount() {
    
    this.checkServerPing();
    if (ping.timerPing === undefined) {

      ping.timerPing = setInterval(() => {
        
        this.checkServerPing();

      }, ping.timerInterval);
    }
  }

  render() {
    
    return (
      <Router>

        <Route extact path="/" render={(props) => <LoginPage authentificationObtained={authentificationObtained} ping={ping} {...props} />} />

        <PioneerRoute path="/home" render={(props) => <HomePage ping={ping} {...props} />} />
        <PioneerRoute path="/pioneer" render={(props) => <PioneerPage authentificationObtained={authentificationObtained} ping={ping} {...props} />} />
        <PioneerRoute path="/variation" render={(props) => <VariationPage authentificationObtained={authentificationObtained} ping={ping} {...props} />} />
        <PioneerRoute path="/resource" render={(props) => <ResourcePage ping={ping} {...props} />} />

        <AdministratorRouter path="/administration" render={(props) => <AdministrationPage authentificationObtained={authentificationObtained} {...props} />} />

      </Router>
    )
  }
}

export default App
