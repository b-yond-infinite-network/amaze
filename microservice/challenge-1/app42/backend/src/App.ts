import {Application, Request, Response} from "express";
import express = require("express");
import Log4js from "log4js";

import UserRouter from "./router/UserRouter";

const logger = Log4js.getLogger(module.filename);
logger.level = "debug";

class App {

    public static creattApp(): any {
        return new App().app;
    }

    private static defaultErrorHandler(error: Error, request: Request, response: Response): void {
        logger.error("[x]]");
        logger.error(error.stack);
        response.status(500).send("Something broke!");
    }

    public app: Application;

    private constructor() {
        this.app = express();
        this.initialize();
        this.setRoutes();
        this.setDefaultErrorHandler();
    }

    private initialize(): void {
        this.app.use(express.json());
        this.app.use(express.urlencoded({extended: false}));
    }

    private setRoutes(): void {
        this.app.use("/users", UserRouter.getRouter());
    }

    private setDefaultErrorHandler(): void {
        this.app.use(App.defaultErrorHandler);
    }

}

export default App;
