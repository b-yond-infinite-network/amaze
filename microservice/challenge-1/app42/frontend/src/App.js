import React, {Component} from 'react';
import {BrowserRouter as Router} from 'react-router-dom'

class App extends Component {
  render() {
    return (
      <Router>
        <React.Fragment>
          <p>test</p>
        </React.Fragment>
      </Router>
    );
  }
}

export default App;
