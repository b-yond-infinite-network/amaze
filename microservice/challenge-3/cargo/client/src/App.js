import React from 'react'
import ToDoList from './ToDoList.js'

function App () {
  const backgroundColors = {
    Default: '#81b71a',
    Cyan: '#37BC9B',
    Green: '#8CC152',
    Red: '#E9573F',
    Yellow: '#F6BB42'
  }
  return (
    <div className="App">
      <div class="top-container" style= {{ backgroundColor: backgroundColors.Cyan }} >
        <h1 style={{ color: backgroundColors.Red, textAlign: 'center' }}>Load that Cargo!!</h1>
        <div style= {{ backgroundColor: backgroundColors.Cyan, fontSize: 22, textAlign: 'center' }}>Add and Delete Cargos using text area.</div>
      </div>
      <ToDoList/>
    </div>
  )
}

export default App
