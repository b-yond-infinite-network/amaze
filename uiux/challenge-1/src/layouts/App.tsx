import { Link } from "@reach/router";
import React, { ReactNode } from "react";
import styles from "./App.module.css";

/**
 * Header component (Main layout)
 */
const Header = () => (
  <header className={styles.header}>
    <img
      alt="Musixmatch"
      className={styles.logo}
      src="//s.mxmcdn.net/site/images/logo_icon.svg"
    />
    <h1>
      <Link to="/">Karaoke needs words</Link>
    </h1>
  </header>
);

type Props = {
  children?: ReactNode;
};

/**
 * Main layout
 */
const App: React.FC<Props> = ({ children }: Props): JSX.Element => (
  <>
    <Header />
    <main>{children}</main>
  </>
);

export default App;
