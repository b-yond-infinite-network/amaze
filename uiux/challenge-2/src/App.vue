<template>
  <div id="app">
    <router-view/>
  </div>
</template>

<style lang="scss">
  @import './styles.scss'
</style>

<style lang="scss">
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2c3e50;

  a {
    text-decoration:none;
    color:inherit;
  }

  .container {
    padding:0px 15px;
    margin:0 auto;

    @media(min-width:768px) {
      max-width:720px;
    }
    @media(min-width:992px) {
      max-width:960px;
    }
    @media(min-width:1200px) {
      max-width:1140px;
    }
  }

  .text-center {
    text-align:center;
  }

  .block-center {
    display:block;
    margin:0 auto;
  }

  .mt-0 {
    margin-top:0px;
  }

  .mb-0 {
    margin-bottom:0px;
  }

  .dz-button {
    padding:0.5rem 0.75rem;
    border:none;
    cursor:pointer;
    background:#2c3e50;
    color:#ffffff;

    &:hover {
      background:#2c3e65;
    }
  }

  .text-uppercase {
    text-transform:uppercase;
  }

  .text-capitalize {
    text-transform:capitalize;
  }
}

/**
 * ----------------------------------------
 * animation scale-up-ver-top
 * ----------------------------------------
 */
.scale-up-ver-top {
	-webkit-animation: scale-up-ver-top 0.5s cubic-bezier(0.390, 0.575, 0.565, 1.000) both;
	        animation: scale-up-ver-top 0.5s cubic-bezier(0.390, 0.575, 0.565, 1.000) both;
}

@-webkit-keyframes scale-up-ver-top {
  0% {
    -webkit-transform: scaleY(0.4);
            transform: scaleY(0.4);
    -webkit-transform-origin: 100% 0%;
            transform-origin: 100% 0%;
  }
  100% {
    -webkit-transform: scaleY(1);
            transform: scaleY(1);
    -webkit-transform-origin: 100% 0%;
            transform-origin: 100% 0%;
  }
}
@keyframes scale-up-ver-top {
  0% {
    -webkit-transform: scaleY(0.4);
            transform: scaleY(0.4);
    -webkit-transform-origin: 100% 0%;
            transform-origin: 100% 0%;
  }
  100% {
    -webkit-transform: scaleY(1);
            transform: scaleY(1);
    -webkit-transform-origin: 100% 0%;
            transform-origin: 100% 0%;
  }
}

/**
 * ----------------------------------------
 * animation scale-down-ver-top
 * ----------------------------------------
 */
.scale-down-ver-top {
	-webkit-animation: scale-down-ver-top 0.5s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;
	        animation: scale-down-ver-top 0.5s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;
}

@-webkit-keyframes scale-down-ver-top {
  0% {
    -webkit-transform: scaleY(1);
            transform: scaleY(1);
    -webkit-transform-origin: 100% 0%;
            transform-origin: 100% 0%;
  }
  100% {
    -webkit-transform: scaleY(0.3);
            transform: scaleY(0.3);
    -webkit-transform-origin: 100% 0%;
            transform-origin: 100% 0%;
  }
}
@keyframes scale-down-ver-top {
  0% {
    -webkit-transform: scaleY(1);
            transform: scaleY(1);
    -webkit-transform-origin: 100% 0%;
            transform-origin: 100% 0%;
  }
  100% {
    -webkit-transform: scaleY(0.3);
            transform: scaleY(0.3);
    -webkit-transform-origin: 100% 0%;
            transform-origin: 100% 0%;
  }
}
</style>
