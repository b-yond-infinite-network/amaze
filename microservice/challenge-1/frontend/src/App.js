import React, { Component } from 'react';
import './App.css';
import RetwisjApp from './component/RetwisjApp';

class App extends Component {
  render() {
    return (
      <div className="container">
        <RetwisjApp />
      </div>
    );
  }
}

export default App;
