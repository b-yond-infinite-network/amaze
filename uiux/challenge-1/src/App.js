// Author: Anthony Guevara
// E-mail: anthony.guev@gmail.com
// Date:  May 27, 2018 - version 1.0.0 - Initial working app
//        September 3, 2018 - version 2.0.0 - Refactor App.js render
//            function
//            - Add redux state management
// Challenge 1 - Karaoke needs words

import React, { Component } from 'react';
import './App.css';
import Navbar from './components/Navbar'
import SearchHeader from './components/SearchHeader'
import FormTextField from './components/FormTextField'
import SongPanel from './components/SongPanel'
import ArtistPanel from './components/ArtistPanel'
import SongListPanel from './components/SongListPanel'

// bootstrap-react
import { FormGroup, Grid } from 'react-bootstrap';
// put any other imports below so that CSS from your
// components takes precedence over default styles.
import 'bootstrap/dist/css/bootstrap.css';
import 'bootstrap/dist/css/bootstrap-theme.css';

// react table
import 'react-table/react-table.css';

// redux
import { connect } from 'react-redux';

// main app
class App extends Component {

  render() {
    // css
    var fullScreen = {
      height: '100vh',
      background: 'linear-gradient(to right, #1a2a6c, #b21f1f, #fdbb2d)'
    }

    return (
        <div style={ fullScreen }>
          <Navbar />
          <Grid>
            <SearchHeader />
            <FormGroup controlId="artistSearch">
              <FormTextField />
            </FormGroup>
            <ArtistPanel/>
            <SongListPanel/>
            <SongPanel/>
          </Grid>
        </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
  }
};

const mapActionsToProps = {
};

export default connect(mapStateToProps, mapActionsToProps)(App);
