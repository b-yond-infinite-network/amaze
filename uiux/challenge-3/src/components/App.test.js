import App from './App';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router'
import configureMockStore from 'redux-mock-store'
import middlewares from './../store'

const initialState = {
    home: {
        tags: ['tag1', 'tag2']
    },
    common: {
        appLoaded: true,
        appName: 'test appName',
        currentUser: {
            username: 'Test User',
            image: ''
        },
        redirectTo: '/'
    }
};

describe('App Component', () => {

    let mockStore;

    let tree;
    let wrapper;
    let render;

    describe('appLoaded === true', () => {

        beforeAll(() => {

            mockStore = configureMockStore(middlewares)(initialState);

            const props = {
            };

            tree = <Provider store={mockStore}>
                <MemoryRouter initialEntries={["/"]}>
                    <App {...props} />
                </MemoryRouter>
            </Provider>;
            wrapper = mount(tree);
            console.log(wrapper.debug());
            render = renderer.create(tree).toJSON();

        });

        it('Should match the snapshot', () => {
            expect(render).toMatchSnapshot();
        });

    });

    describe('appLoaded === false', () => {

        beforeAll(() => {

            initialState.common.appLoaded = false;
            mockStore = configureMockStore(middlewares)(initialState);

            const props = {
            };

            tree = <Provider store={mockStore}>
                <MemoryRouter initialEntries={["/"]}>
                    <App {...props} />
                </MemoryRouter>
            </Provider>;
            wrapper = mount(tree);
            console.log(wrapper.debug());
            render = renderer.create(tree).toJSON();

        });

        it('Should match the snapshot', () => {
            expect(render).toMatchSnapshot();
        });

    });

});