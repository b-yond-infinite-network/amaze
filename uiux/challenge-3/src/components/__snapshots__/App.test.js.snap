// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component appLoaded === false Should match the snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-light"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        test appname
      </a>
      <ul
        className="nav navbar-nav pull-xs-right"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/editor"
            onClick={[Function]}
          >
            <i
              className="ion-compose"
            />
             New Post
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/settings"
            onClick={[Function]}
          >
            <i
              className="ion-gear-a"
            />
             Settings
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/@Test User"
            onClick={[Function]}
          >
            <img
              alt="Test User"
              className="user-pic"
              src=""
            />
            Test User
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`App Component appLoaded === true Should match the snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-light"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        test appname
      </a>
      <ul
        className="nav navbar-nav pull-xs-right"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/editor"
            onClick={[Function]}
          >
            <i
              className="ion-compose"
            />
             New Post
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/settings"
            onClick={[Function]}
          >
            <i
              className="ion-gear-a"
            />
             Settings
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/@Test User"
            onClick={[Function]}
          >
            <img
              alt="Test User"
              className="user-pic"
              src=""
            />
            Test User
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="home-page"
    id="home-page"
  >
    <div
      className="banner"
      id="banner"
    >
      <div
        className="container"
      >
        <h1
          className="logo-font"
        >
          test appname
        </h1>
        <p>
          A place to share your knowledge.
        </p>
      </div>
    </div>
    <div
      className="container page"
    >
      <div
        className="row"
      >
        <div
          className="col-md-9"
          id="main-view"
        >
          <div
            className="feed-toggle"
          >
            <ul
              className="nav nav-pills outline-active"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href=""
                  onClick={[Function]}
                >
                  Global Feed
                </a>
              </li>
            </ul>
          </div>
          <div
            className="article-preview"
          >
            Loading...
          </div>
        </div>
        <div
          className="col-md-3"
        >
          <div
            className="sidebar"
          >
            <p>
              Popular Tags
            </p>
            <div
              className="tag-list"
              id="tag-list"
            >
              <a
                className="tag-default tag-pill"
                href=""
                onClick={[Function]}
              >
                tag1
              </a>
              <a
                className="tag-default tag-pill"
                href=""
                onClick={[Function]}
              >
                tag2
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
